test/chemistry/test_app_mgse.py

```diff
@@ -179,6 +179,46 @@ def cb_create_solver(num_particles, num_orbitals,
         result = mgse.compute_energy(cb_create_solver)
         self.assertAlmostEqual(result.energy, -7.882, places=3)
 
+    def test_mgse_callback_vqe_uccsd_z2_nosymm(self):
```

Comment:
This looks like a pretty time-consuming test for a change in the default argument. As I understand it, this is the typical use-case where this would come up but wouldn't it make sense here to just check if the Z2Symmetries object is not None instead of running a full simulation?
The motivation being less end-to-end testing and faster tests 🙂

Reply:
It tests that the result when there is no symmetry detected, when there are none in the problem, i.e the main operator, aux operators give the same result. Yes the Z2Symmetry is also used since it gets passed to HartreeFock and UCCSD, and part of the failure was it did not get set to any empty set. The only other choice I had was to use use the core object and check the operator equality and Z2 object it outputs - but there are other internal variables that are stored that go into processing the final outcome from the algorithm that using that would not be enough.

Reply:
It occurred to me that I can check the Z2_symmetery passed in the callback to ensure its empty and then check the operators & final result using a classical solver. This should speed this up a bit.